Include.psm1

```diff
@@ -519,60 +519,54 @@ function Set-Stat {
     $Stat
 }
 
-function Get-Stat { 
+function Get-Stat {
     [CmdletBinding()]
     param(
         [Parameter(Mandatory = $false)]
-        [String[]]$Name = (Get-ChildItem "Stats" -ErrorAction Ignore | Select-Object -ExpandProperty BaseName)
+        [String]$Name
     )
 
-    if (-not (Test-Path "Stats" -PathType Container)) { 
-        New-Item "Stats" -ItemType "directory" -Force | Out-Null
-    }
-
     if ($Global:Stats -isnot [PSCustomObject]) { 
         $Global:Stats = [PSCustomObject]@{ }
     }
 
-    Get-ChildItem -Path "Stats" -File | Where-Object { $Name -eq $_.BaseName -and ".txt" -eq $_.Extension } | ForEach-Object { 
-        $BaseName = $_.BaseName
-        $FullName = $_.FullName
+    if ($Global:Stats.$Name) { return $Global:Stats.$Name }
 
-        if (-not $Global:Stats.$BaseName) { 
+    if ($Name) {
+        # Return single requested stat
+        if (Test-Path "Stats\$Name.txt" -ErrorAction SilentlyContinue) { 
             try { 
-                $Global:Stats | Add-Member @{ 
-                    $BaseName = $_ | Get-Content -ErrorAction Stop | ConvertFrom-Json -ErrorAction Stop | ForEach-Object { 
-                        [PSCustomObject]@{ 
```

Comment:
Please enforce the stat structure.

Reply:
> Please enforce the stat structure.

for what benefit? Get-Stat is one of the mostused functions and should therefore be as fast as possible.

Reply:
Stats has a controlled structure so you should not read in uncontrolled data directly. 

It is better to just focus on the bottle necks. The disk reading will be the part that was slowing it down. 

I'll sort it out shortly.